### ISSUE TITLE
Not working with latest Odoo Enterprise debian distribution due to python3-suds package version difference.

### ISSUE THREAD

**Issue by mladeny** [sodexis/avatax_connector#6]
When confirming a quotation/sale order, I receive the following error message:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/suds/transport/http.py", line 80, in send
    fp = self.u2open(u2request)
  File "/usr/lib/python3/dist-packages/suds/transport/http.py", line 127, in u2open
    return url.open(u2request, timeout=tm)
  File "/usr/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.5/urllib/request.py", line 510, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 590, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 500: Internal Server Error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/suds/client.py", line 747, in send
    reply = self.options.transport.send(request)
  File "/usr/lib/python3/dist-packages/suds/transport/https.py", line 66, in send
    return HttpTransport.send(self, request)
  File "/usr/lib/python3/dist-packages/suds/transport/http.py", line 90, in send
    raise TransportError(e.msg, e.code, e.fp)
suds.transport.TransportError: Internal Server Error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 936, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 935, in call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 923, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/delivery/models/sale_order.py", line 54, in action_confirm
    res = super(SaleOrder, self).action_confirm()
  File "/usr/lib/python3/dist-packages/odoo/addons/website_quote/models/sale_order.py", line 217, in action_confirm
    res = super(SaleOrder, self).action_confirm()
  File "/usr/lib/python3/dist-packages/odoo/addons/sale_timesheet/models/sale_order.py", line 62, in action_confirm
    result = super(SaleOrder, self).action_confirm()
  File "/usr/lib/python3/dist-packages/odoo/addons/sale_subscription/models/sale_order.py", line 129, in action_confirm
    res = super(SaleOrder, self).action_confirm()
  File "/usr/lib/python3/dist-packages/odoo/addons/project_forecast_sale/models/sale_order.py", line 8, in action_confirm
    return super(SaleOrder, self.with_context(default_allow_forecast=True)).action_confirm()
  File "/usr/lib/python3/dist-packages/odoo/addons/avatax_connector/models/sale_order.py", line 243, in action_confirm
    self.compute_tax()
  File "/usr/lib/python3/dist-packages/odoo/addons/avatax_connector/models/sale_order.py", line 216, in compute_tax
    currency_id=self.currency_id).TotalTax
  File "/usr/lib/python3/dist-packages/odoo/addons/avatax_connector/models/account_tax.py", line 78, in _get_compute_tax
    location_code, currency_code, partner.vat_id or None, is_override)
  File "/usr/lib/python3/dist-packages/odoo/addons/avatax_connector/models/avalara_api.py", line 197, in get_tax
    result = self.get_result(self.taxSvc, self.taxSvc.service.GetTax, request)
  File "/usr/lib/python3/dist-packages/odoo/addons/avatax_connector/models/avalara_api.py", line 96, in get_result
    result = operation(request)
  File "/usr/lib/python3/dist-packages/suds/client.py", line 566, in __call__
    return client.invoke(args, kwargs)
  File "/usr/lib/python3/dist-packages/suds/client.py", line 705, in invoke
    result = self.send(soapenv)
  File "/usr/lib/python3/dist-packages/suds/client.py", line 752, in send
    return self.process_reply(content, e.httpcode, tostr(e))
  File "/usr/lib/python3/dist-packages/suds/client.py", line 814, in process_reply
    raise WebFault(fault, replyroot)
suds.WebFault: Server raised fault: 'Instance validation error: '' is not a valid value for TaxOverrideType.'
```

I tracked this down to a package version difference in Odoo Enterprise's debian distribution.

Odoo Enterprise debian distribution installs python3-suds version 0.7, whereas the source code version of Odoo Enterprise has version 0.6 in requirements.txt.

Manually forcing a downgrade on our server for python3-suds to 0.6 resolves it. 
Had to patch it with pip, since python3-suds version 0.6 is not available in the Ubuntu 16.04 repositories. 

It appears the issue has to do with the soap/xml marshalling of the python objects, where TaxOverrideType is being set to '' instead of None.

Is it possible to port avatax_connector to work with python3-suds version 0.7?

**Comment by mladeny** [comment]
Fix for Ubuntu 16.04 and 18.04 Odoo enterprise installation using .deb (debian) package.

Remove suds jurko (.deb Odoo package installs suds-jurko 0.7 dev, which causes issues with avatax_connector)

```
sudo pip remove suds-jurko
```

Install suds-jurko version 0.6 needed by avatax
```
sudo pip install suds-jurko==0.6
```


Restart Odoo

```
sudo systemctl restart odoo
```

You can now confirm sales orders

**Comment by jorenvo** [comment]
Thank you for your detailed report and for figuring out the problem! Indeed Debian Stretch comes with suds-jurko 0.7. It's a bit strange because this version is not published on pypi.org. I think it's possible it's still in development and the author will publish it when it's ready. Regardless, I worked around the issues I detected in https://github.com/fhe-odoo/avatax_connector/commit/0dc95cb67e3bda2fcf54afd677390c73dc93e728. In case you decide to go back to `python3-suds` and run into additional issues let us know.
